haystack/components/rankers/sentence_transformers_similarity.py

```diff
@@ -0,0 +1,280 @@
+            A string to add at the beginning of each document before ranking. You can use it to prepend the document
+            with an instruction, as required by embedding models like `bge`.
+        :param meta_fields_to_embed:
+            List of metadata fields to embed with the document.
+        :param embedding_separator:
+            Separator to concatenate metadata fields to the document.
+        :param scale_score:
+            If `True`, scales the raw logit predictions using a Sigmoid activation function.
+            If `False`, disables scaling of the raw logit predictions.
+        :param score_threshold:
+            Use it to return documents with a score above this threshold only.
+        :param model_kwargs:
+            Additional keyword arguments for `AutoModelForSequenceClassification.from_pretrained`
+            when loading the model. Refer to specific model documentation for available kwargs.
+        :param tokenizer_kwargs:
+            Additional keyword arguments for `AutoTokenizer.from_pretrained` when loading the tokenizer.
+            Refer to specific model documentation for available kwargs.
+        :param config_kwargs:
+            Additional keyword arguments for `AutoConfig.from_pretrained` when loading the model configuration.
+        :param backend:
+            The backend to use for the Sentence Transformers model. Choose from "torch", "onnx", or "openvino".
+            Refer to the [Sentence Transformers documentation](https://sbert.net/docs/sentence_transformer/usage/efficiency.html)
+            for more information on acceleration and quantization options.
+        :param batch_size:
+            The batch size to use for inference. The higher the batch size, the more memory is required.
+            If you run into memory issues, reduce the batch size.
+
+        :raises ValueError:
+            If `top_k` is not > 0.
+        """
+        torch_and_sentence_transformers_import.check()
+
+        if top_k <= 0:
+            raise ValueError(f"top_k must be > 0, but got {top_k}")
+
+        self.model_name_or_path = str(model)
+        self.model = None
+        self.query_prefix = query_prefix
+        self.document_prefix = document_prefix
+        self.device = ComponentDevice.resolve_device(device)
+        self.top_k = top_k
+        self.token = token
+        self.meta_fields_to_embed = meta_fields_to_embed or []
+        self.embedding_separator = embedding_separator
+        self.scale_score = scale_score
+        self.score_threshold = score_threshold
+        self.model_kwargs = model_kwargs
+        self.tokenizer_kwargs = tokenizer_kwargs
+        self.config_kwargs = config_kwargs
+        self.backend = backend
+        self.batch_size = batch_size
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        return {"model": self.model_name_or_path}
+
+    def warm_up(self):
```

Comment:
In the spirit of the stricter type checking :D 
```suggestion
    def warm_up(self) -> None:
```